resources/assets/react/screens/Profile/AssociativeCareer.js

```diff
@@ -0,0 +1,124 @@
+/**
+ * Display all associations of a given user.
+ *
+ * @author Corentin Mercier <corentin@cmercier.fr>
+ *
+ * @copyright Copyright (c) 2019, SiMDE-UTC
+ * @license GNU GPL-3.0
+ */
+
+import React from 'react';
+import { connect } from 'react-redux';
+import { NavLink } from 'react-router-dom';
+import AssoCard from '../../components/AssoCard';
+
+import actions from '../../redux/actions';
+
+@connect(store => ({
+	config: store.config,
+	semesters: store.getData(['semesters']),
+	semestersFetched: store.isFetched(['semesters']),
+	roles: store.getData('roles'),
+	rolesIsFetching: store.isFetching('roles'),
+	rolesFetched: store.isFetched('roles'),
+}))
+class AssociativeCareerScreen extends React.Component {
+	constructor(props) {
+		super(props);
+
+		this.state = {
+			associativeSemesters: {},
+		};
+	}
+
+	componentDidMount() {
+		const {
+			user: { name },
+			dispatch,
+			rolesFetched,
+			semesters,
+		} = this.props;
+		const { associativeSemesters } = this.state;
+
+		if (!rolesFetched) {
+			dispatch(actions.roles.all());
+		}
+
+		dispatch(actions.config({ title: `${name} - Mon Parcours` }));
+
+		semesters
+			.slice()
+			.reverse()
+			.forEach(semester => {
+				if (!associativeSemesters[semester.id]) {
+					actions.user.assos.all({ semester: semester.id }).payload.then(({ data }) => {
+						if (data.length > 0) {
+							this.addNewAssociativeSemester(semester.id, data);
+						}
+					});
+				}
+			});
+	}
+
+	componentDidUpdate() {
+		const {
+			user: { name },
+			dispatch,
+			rolesFetched,
+		} = this.props;
+
+		if (!rolesFetched) {
+			dispatch(actions.roles.all());
+		}
+
+		dispatch(actions.config({ title: `${name} - Mon Parcours` }));
+	}
+
+	addNewAssociativeSemester(semester_id, assos) {
+		const { associativeSemesters } = this.state;
+		const objectToAssign = {};
+		objectToAssign[semester_id] = assos;
+		this.setState({
```

Comment:
Here you can simply do:
`this.setState(prevState => { prevState.associativeSemesters[semester_id] = assos; return prevState; })